components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -2453,6 +2453,17 @@ public String getSwagger12Definition(APIIdentifier apiId) throws APIManagementEx
 		return apiJSON.toJSONString();
 	}
 
+    /**
+     * Returns the all the Consumer keys of applications which are subscribed to the given API
+     *
+     * @param apiIdentifier APIIdentifier
+     * @return a String array of ConsumerKeys
+     * @throws APIManagementException
+     */
+    public String[] getConsumerKeys(APIIdentifier apiIdentifier) throws APIManagementException {
+
+        return apiMgtDAO.getConsumerKeys(apiIdentifier);
```

Comment:
I might have misread this. The way this has already been implemented is all good. Sorry for the noise.